.github/workflows/run_functional_product_tests.yaml

```diff
@@ -10,12 +10,70 @@ on:
   workflow_dispatch:
 
 jobs:
-  product-tests:
-    name: Product Tests
+  # Build a matrix of product param combinations for testing.
+  build-matrix:
+    name: Build Matrix
     runs-on: windows-latest
     defaults:
       run:
         shell: powershell
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Generate Matrix
+        id: generate-matrix
+        # The testing params are stored as a string in a CSV-like format.
+        # Pipes separate products, and commas separate key/value pairs.
+        # key1=value1,key2=value2|key3=value3,key4=value4...
+        # This job parses out the key-value pairs and puts them in an array
+        # that will be used as a matrix for testing each product combination.
+        run: |
+          $params = $env:TestParams
+          # Powershell doesn't build arrays in the format that a matrix
+          # wants, so this builds it using a string instead.  It's ugly.
+          $productsAsList = "["
+          # Split into products
+          $products = $params.split("|")
+          foreach ($product in $products)
+          {
+            # Since commas will separate products in the matrix,
+            # this will use semi-colons to separate key-value pairs.
+            $product = $product.replace(',', ';')
+            $productsAsList += '"'
+            # This trim is necessary because the final product ends with
+            # a newline that causes problems.
+            $productsAsList += $product.Trim()
+            $productsAsList += '",'
+          }
+          $productsAsList = $productsAsList.Substring(0, $productsAsList.Length-1)
+          $productsAsList += ']'
+          # Store the string "array" where it can be accessed as an output.
+          echo products=$productsAsList >> $env:GITHUB_OUTPUT
+        env:
+          TestParams: ${{ secrets.NIGHTLY_TEST_BUILD_PARAMS }}
+    outputs:
+      matrix: ${{ steps.generate-matrix.outputs.products }}
+      # The purpose of this title is simply to mask the params in the
+      # test job names.
+      title: Product
+  # Use ScubaGear to test each service.
+  # This job runs once per param combination in the matrix.
+  test-products:
+    name: Test ${{ needs.build-matrix.outputs.title }}
```

Comment:
I expected to see a more informative title (i.e., Testing Teams).  

![image](https://github.com/cisagov/ScubaGear/assets/7025068/696dde90-d742-4850-be2d-d08d5273bfc0)

Reply:
Yes, I concur, too.  But I don't know how to change the name of a job based upon values are that created in its steps.  IOW, the title is created first, the steps are executed second, and the names (aliases) of the products are parsed out third.  If you have ideas on how to improve, I am all ears.